src/wasm/nxt_wasm.c

```diff
@@ -236,6 +247,33 @@ nxt_wasm_setup(nxt_task_t *task, nxt_process_t *process,
     fh[NXT_WASM_FH_REQUEST_END].func_name = c->request_end_handler;
     fh[NXT_WASM_FH_RESPONSE_END].func_name = c->response_end_handler;
 
+    /* Get any directories to pass through to the WASM module */
+    if (c->access != NULL) {
+        dirs = nxt_conf_get_object_member(c->access, &filesystem_str, NULL);
+    }
+
+    n = (dirs != NULL) ? nxt_conf_object_members_count(dirs) : 0;
+    if (n == 0) {
+        goto out_init;
+    }
+
+    nxt_wasm_ctx.dirs = nxt_zalloc((n + 1) * sizeof(char *));
+    if (nxt_slow_path(nxt_wasm_ctx.dirs == NULL)) {
+        return NXT_ERROR;
+    }
+
+    for (i = 0; i < n; i++) {
+        nxt_str_t         str;
+        nxt_conf_value_t  *value;
+
+        value = nxt_conf_get_array_element(dirs, i);
+        nxt_conf_get_string(value, &str);
+
+        nxt_wasm_ctx.dirs[i] = nxt_zalloc(str.length + 1);
+        nxt_cpymem(nxt_wasm_ctx.dirs[i], str.start, str.length);
+    }
+
+out_init:
     err = nxt_wops->init(&nxt_wasm_ctx);
     if (err) {
         exit(EXIT_FAILURE);
```

Comment:
What is that `+1` allocation used for?

Reply:
Another comment that went AWOL

Long story short; so we have terminating NULL entry...